### ISSUE TITLE
Different results for dl4j versions 0.8.0 and 0.9.1

### ISSUE THREAD

**Issue by MariyaF** [deeplearning4j/deeplearning4j#5046]
Issue Description

I switched from the library version 0.8.0 to the version 1.0.0 alpha. Parameters of ANN did not change. Retrained with new version. My RMSE change from 0.1 to 0.6.
Why the results are so different for different versions?

Version Information

Deeplearning4j version from 0.8.0 to 1.0.0 alpha
platform information (OS, etc) windows 10
Contributing

Topology for 0.8.0

`
private MultiLayerConfiguration getDeepDenseLayerNetworkConfiguration() {
final int numHiddenNodes = 100;

    return new NeuralNetConfiguration.Builder()
            .seed(SEED)
            .iterations(ITERATIONS)
            .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
            .learningRate(LEARNING_RATE)
            .weightInit(WeightInit.XAVIER)
            .updater(Updater.NESTEROVS)
            .momentum(0.9)
            .list()
            .layer(0, new GravesLSTM.Builder().nIn(NUM_INPUTS).nOut(numHiddenNodes).activation(Activation.HARDTANH).build())
            .layer(1, new GravesLSTM.Builder().nIn(numHiddenNodes).nOut(numHiddenNodes).activation(Activation.HARDTANH).build())
            .layer(2, new RnnOutputLayer.Builder(LossFunctions.LossFunction.MSE).activation(Activation.SIGMOID).nIn(numHiddenNodes).nOut(NUM_OUTPUTS).build())
            .pretrain(false)
            .backprop(true)
            .build();
}`
Topology for 1.0.0 alpha

`
private MultiLayerConfiguration getDeepDenseLayerNetworkConfiguration() {
final int numHiddenNodes = 100;

    return new NeuralNetConfiguration.Builder()
            .seed(SEED)
            .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
            .weightInit(WeightInit.XAVIER)
            .updater(new Nesterovs(LEARNING_RATE, 0.9))
            .list()
            .layer(0, new GravesLSTM.Builder().nIn(NUM_INPUTS).nOut(numHiddenNodes).activation(Activation.HARDTANH).build())
            .layer(1, new GravesLSTM.Builder().nIn(numHiddenNodes).nOut(numHiddenNodes).activation(Activation.HARDTANH).build())
            .layer(2, new RnnOutputLayer.Builder(LossFunctions.LossFunction.MSE).activation(Activation.SIGMOID).nIn(numHiddenNodes).nOut(NUM_OUTPUTS).build())
            .pretrain(false)
            .backprop(true)
            .build();
}`

**Comment by AlexDBlack** [comment]
Please don't open duplicate issues. https://github.com/deeplearning4j/deeplearning4j/issues/5037

If you think the issue wasn't sufficiently resolved, you can ask in the original issue for it to be re-opened.
But - as per my post in the original issue - you still haven't given us sufficient information (notably, the dataset used and how you are calculating RMSE) to actually reproduce this.

**Comment by mxmaster-focus** [comment]
123
